boa_engine/src/vm/code_block.rs

```diff
@@ -252,6 +250,33 @@ impl CodeBlock {
             .rev()
             .find(|(_, handler)| handler.contains(pc))
     }
+
+    pub(crate) fn constant_string_expect(&self, index: usize) -> JsString {
+        if let Constant::String(value) = &self.constants[index] {
+            return value.clone();
+        }
+
+        panic!("there should be a string constant at index {index}")
+    }
```

Comment:
A very minor optimization would be to do a `get` on the slice instead of directly accessing it via the array index notation. Then do `if let Some(Constant::String(value))`.
That way we avoid the panic branch on the bounds check and only have one panic in this function.